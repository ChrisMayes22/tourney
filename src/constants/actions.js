export const INITIALIZE_CHARACTER = 'INITIALIZE_CHARACTER';
export const REMOVE_CHARACTER = 'REMOVE_CHARACTER';
export const REMOVE_LOWEST_SCORE = 'REMOVE_LOWEST_SCORE';
export const RATE_CHARACTER = 'RATE_CHARACTER';
export const SUBMIT_CHARACTER_RATINGS = 'SUBMIT_CHARACTER_RATINGS';
export const SET_PLAYERS = 'SET_PLAYERS';
export const RESET_FOR_FINALS = 'RESET_FOR_FINALS';
export const CHOOSE_WINNER = 'CHOOSE_WINNER';

export const initializeCharacter = function(imageUrl, id){
    return {type: INITIALIZE_CHARACTER, 
        payload: {image: imageUrl, id}}
}
export const removeCharacter = function(id){
    console.log('Remove Character Dispatched')
    console.log(REMOVE_CHARACTER)
    return {type: REMOVE_CHARACTER, id}
}
export const removeLowestScore = function(){
    return {type: REMOVE_LOWEST_SCORE};
}
export const rateCharacter = function(points, rowId, character){
    return {type: RATE_CHARACTER, payload:{
        points, rowId, character}}
}
export const submitCharacterRatings = function(playersArray){
    return {type: SUBMIT_CHARACTER_RATINGS, payload:{
        players: playersArray
    }};
};
export const setPlayers = function(num){
    return {type: SET_PLAYERS, number: num}
};
export const resetForFinals = function(){
    return {type: RESET_FOR_FINALS}
}
export const chooseWinner = function(character){
    return {type: CHOOSE_WINNER, character}
}

