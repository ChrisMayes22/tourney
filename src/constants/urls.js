export const VOTING_PAGE = '/voting';
export const ELIMINATION_PAGE = '/elimination';
export const NOT_ENOUGH_CHARACTERS = '/not-enough-characters';
export const UPLOAD_PAGE = '/upload';
export const WINNER_PAGE = '/winner';
export const HOME = '/';