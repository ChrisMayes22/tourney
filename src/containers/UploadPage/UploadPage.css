.flexContainer__col{
    display: flex;
    margin: 5px auto;
    flex-direction: column;
    width: 95vw;
    height: 40vh;
    align-items: center;
    background-color: #a63939;
    background-color: #a63939;
    color: rgb(253, 211, 250);
    font-size: 1.3rem;
    border-radius: 10px;
}

h3{
    margin: 1.2rem 0 0.8rem 0;
}

h1, h5{
    margin: 0.3rem;
}

.inputContent{
    text-align: center;
    margin: 0;
}

.inputContent button{
    display: block;
    margin: 1.2rem auto;
}

[type="text"]{
    background-color: rgb(255, 239, 254);
    color: #a63939;
    text-align: center;
    height: 5vh;
    width: 40vw;
    font-size: 1.2rem;
    border: none;
    border-radius: 5px;
}

.grid {
    margin: 5px auto;
    display: grid;
    grid-template-rows: 15vh;
    grid-template-columns: 19vw 19vw 19vw 19vw 19vw;
    grid-template-areas: "header header header header header";
    grid-auto-rows: 30vh;
    grid-auto-columns: minmax(auto, 19vw);
    background-color: rgb(253, 211, 250);
    justify-items: center;
    align-items: center;
}

.imageHeader{
    text-align: center;
    background-color: rgb(253, 211, 250);
    color: #a63939;
    grid-area: header;
    object-fit: contain;
}

.refImg{
    height: auto;
    max-height: 28vh;
    max-width: 18vw;
    border-radius: 15px;
    box-shadow: 5px 4px 3px rgba(0, 0, 0, 0.7);
}

.modalHeader{
    margin: 3rem 0 0 0;
}

.modalSizing{
    top: 20vh;
    left: 10vw;
    height: 43vh;
    width: 80vw;
}

.modalFlexContainer{
    margin: 0 auto;
    display: flex;
    justify-content: space-around;
    height: 25vh;
    width: 70vw;
}

.modalImage{
    border-radius: 8px;
    background-color: rgb(44, 9, 9);
    max-height: 20vh;
    max-width: 30vw;
    box-shadow: 2px 2px 10px rgb(0, 0, 0);
}


