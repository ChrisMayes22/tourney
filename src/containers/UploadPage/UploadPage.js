import React, { Component } from 'react';
import classes from './UploadPage.css';
import SubmitButton from '../../components/SubmitButton/SubmitButton';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import {initializeCharacter, removeCharacter, removeLowestScore} from '../../constants/actions'
import * as urls from '../../constants/urls';

export class UploadPage extends Component {

    state = {
        imageUrl: '',
        modalDisplayToggle: true
    }

    imageUrlHandler = (event) => {
        const imageUrl = event.target.value;
        this.setState({
            imageUrl: imageUrl
        })
    }

    modalDisplay = (this.props.characters.findIndex(character => character.hadTurn)+1);

    modalDisplayHandler = () => {
        const toggled = !this.state.modalDisplayToggle;
        this.setState({
            modalDisplayToggle: toggled
        })
    }
    
    render(){

        return(
            <React.Fragment>
                {(this.modalDisplay * this.state.modalDisplayToggle)? <div className={classes.modalBackground} id='modal'>
                        <aside className={classes.modal} >
                        <h5 className= {classes.modalHeader}>
                            DO YOU WANT TO ADD ANOTHER ROUND OF CHARACTERS... <br/>
                            ...OR MOVE ON TO THE SORTING?
                        </h5>
                            <br/>
                            <SubmitButton 
                                children={'Add another!'}
                                clicked={this.modalDisplayHandler}/>
                            <Link to={urls.ELIMINATION_PAGE} onClick={this.props.onEliminate}>
                                <SubmitButton children={'Move on to the finals.'}/>
                            </Link>
                        </aside>
                    </div> : null}

                <div className={classes.flexContainer__col}>
                    <section className={classes.inputContent}>
                        <h3>{`Upload ${this.props.players} images that will be competing, (using a url)...<br/>
                        ...Or begin the game.`}</h3>
                        <input 
                            type="text"
                            onChange={this.imageUrlHandler}
                            default="image-url">
                        </input>
                            <SubmitButton 
                                children={"Upload"}
                                id={"uploadButton"}
                                clicked={() => this.props.onUpload(this.state.imageUrl,
                                                        `url=${String(this.state.imageUrl)}${Math.floor(Math.random()*1000000)}`)}
                            />
                        <Link 
                            to={this.props.characters.filter(el => !el.hadTurn).length === this.props.players ? urls.VOTING_PAGE : urls.NOT_ENOUGH_CHARACTERS} 
                            id="beginGameButton">
                                <SubmitButton children={"Begin Game"}/>
                        </Link>
                    </section>
                    <section className={classes.grid} id='grid'>
                        <header className={classes.imageHeader}>
                            <h1>Uploaded Images</h1>
                            <h5>Click Image to Remove</h5>
                        </header>
                        {this.props.characters.map(el =>{
                            return (
                                <div className={classes.imageContainer} key={el.id}>
                                    <img 
                                        className={classes.refImg} 
                                        src={el.imageUrl} 
                                        onClick={() => this.props.onRemove(el.id)}
                                        alt={"One of the competing items"}/>
                                </div>
                            )
                        })}
                    </section>
                </div>
            </React.Fragment>
        )
    } 
}

const mapDispatchToProps = dispatch => {
    return {
        onUpload: (imageUrl, id) => dispatch(initializeCharacter(imageUrl, id)),
        onRemove: (id) => dispatch(removeCharacter(id)),
        onEliminate: () => dispatch(removeLowestScore())
    }
}

const mapStateToProps = state => {
    return {
        characters: state.characters,
        players: state.players
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(UploadPage);