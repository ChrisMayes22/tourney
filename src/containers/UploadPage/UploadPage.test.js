import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { UploadPage } from './UploadPage';
import * as urls from '../../constants/urls';

var wrapper = null;

describe('When unconnected component first renders', function(){
    let props = {characters: [], players: 4}
    wrapper = shallow(<UploadPage {...props}/>)
    let component = wrapper.instance();

    test('UploadPagerenders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    })
    describe('state initializes as expected', function(){
        test('initializes state w/ empty imageUrl', function(){
            expect(component.state.imageUrl).toBe('');
        })
        test('initializes state w/ modalDisplayToggle set to true', function(){
            expect(component.state.modalDisplayToggle).toBe(true);
        })
    })
    describe('conditionally rendered elements behave as expected', function(){
        describe('modal behaves as exepected', function(){
            test('modal does not render if the characters array is empty.', function(){
                expect(wrapper.exists('#modal')).toBe(false);
            })
            test('modal does not render if the characters array contains at least one character but its hadTurn property is false.', function(){
                props = {characters: [{hadTurn: false, id: 'foo$id'}], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);
        
                expect(wrapper.exists('#modal')).toBe(false);
            })
            test('modal DOES render if the characters array contains at least one character and its hadTurn property is true.', function(){
                props = {characters: [{hadTurn: true, id: 'foo$id'}], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);
        
                expect(wrapper.exists('#modal')).toBe(true);
            })
        })
        describe('`to attribute` of `#beginGameButton` renders as expected', function(){
            test('Given that there are fewer characters than players, the Begin Game to attribute will be urls.NOT_ENOUGH_CHARACTERS.', function(){
                let props = {characters: [], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);
        
                expect(wrapper.exists(`[to="${urls.NOT_ENOUGH_CHARACTERS}"]`)).toBe(true);
                expect(wrapper.exists(`[to="${urls.VOTING_PAGE}"]`)).not.toBe(true);
            })
            test('Given that there are more characters w/ hadTurn:false than players, the Begin Game to attribute will be urls.NOT_ENOUGH_CHARACTERS.', function(){
                let props = {characters: [{hadTurn: false, id: 'foo$id'},{hadTurn: false, id: 'foo$id1'},
                            {hadTurn: false, id: 'foo$id2'},{hadTurn: false, id: 'foo$id3'}], players: 3}
                wrapper = shallow(<UploadPage {...props}/>);
        
                expect(wrapper.exists(`[to="${urls.NOT_ENOUGH_CHARACTERS}"]`)).toBe(true);
                expect(wrapper.exists(`[to="${urls.VOTING_PAGE}"]`)).not.toBe(true);
            })
            describe('Given that the number of characters w/ hadTurn:false === players, the Begin Game to attribute will be urls.VOTING_PAGE.', function(){
                test('when number of characters === players, and all characters have hadTurn:false', function(){
                    let props = {characters: [{hadTurn: false, id: 'foo$id'},{hadTurn: false, id: 'foo$id1'},
                                            {hadTurn: false, id: 'foo$id2'},{hadTurn: false, id: 'foo$id3'}], 
                                players: 4};
                    wrapper = shallow(<UploadPage {...props}/>);
        
                    expect(wrapper.exists(`[to="${urls.NOT_ENOUGH_CHARACTERS}"]`)).not.toBe(true);
                    expect(wrapper.exists(`[to="${urls.VOTING_PAGE}"]`)).toBe(true);
                })
                test('when number of characters !== players, but number w/ hadTurn:false === players', function(){
                    let props = {characters: [{hadTurn: false, id: 'foo$id'},{hadTurn: false, id: 'foo$id1'},{hadTurn: false, id: 'foo$id2'},
                                            {hadTurn: false, id: 'foo$id3'},{hadTurn: true, id: 'foo$id4'}], 
                                players: 4};
                    wrapper = shallow(<UploadPage {...props}/>);
        
                    expect(wrapper.exists(`[to="${urls.NOT_ENOUGH_CHARACTERS}"]`)).not.toBe(true);
                    expect(wrapper.exists(`[to="${urls.VOTING_PAGE}"]`)).toBe(true);
                })
            })
        })
        describe('Characters map renders properly', function(){
            test('when characters map renders, the number of images === characters.length', function(){
                let props = {characters: [{hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'},
                                            {hadTurn: false, id: 'foo$id1', imageUrl: 'foo$image'}], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);

                expect(wrapper.find('[alt="One of the competing items"]')).toHaveLength(2);
            })
        })
    })
})
describe('When unconnected component props or state update', function(){
    describe('When state.modalDisplayToggle toggles', function(){
        test('When UploadPage.modalDisplayHandler is invoked, modal render toggles.', function(){
            let props = {characters: [{hadTurn: true, id: 'foo$id'}], players: 4}
            let wrapper = shallow(<UploadPage {...props}/>)
            const component = wrapper.instance();
    
            expect(wrapper.exists('#modal')).toBe(true);

            component.modalDisplayHandler()
            expect(wrapper.exists('#modal')).toBe(false);
            
            component.modalDisplayHandler()
            expect(wrapper.exists('#modal')).toBe(true);
        })
    })
    describe('When props.characters recieves changes.', function(){
        describe('When a new character is added', function(){
            test('The characters map length updates to === characters.length', function(){
                let props = {characters: [{hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'}], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);
    
                wrapper.setProps({characters:   [{hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'}, 
                                                {hadTurn: false, id: 'foo$id1', imageUrl: 'foo$image'}], 
                                players: 4});
                                
                expect(wrapper.find('[alt="One of the competing items"]')).toHaveLength(2);
            })
        })
        describe('When a character is removed', function(){
            test('The characters map length updates to === characters.length', function(){
                let props = {characters: [{hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'}], players: 4}
                wrapper = shallow(<UploadPage {...props}/>);
    
                wrapper.setProps({characters: [], players: 4});
                                
                expect(wrapper.find('[alt="One of the competing items"]')).toHaveLength(0);
            })
        })
        
    })
})

    
