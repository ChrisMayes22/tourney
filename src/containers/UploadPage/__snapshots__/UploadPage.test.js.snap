// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When unconnected component first renders UploadPagerenders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadPage
    characters={Array []}
    players={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <div>
          <section>
            <h3>
              Upload 4 images that will be competing, (using a url)...
              <br />
              ...Or begin the game.
            </h3>
            <input
              default="image-url"
              onChange={[Function]}
              type="text"
            />
            <submitButton
              clicked={[Function]}
              id="uploadButton"
            >
              Upload
            </submitButton>
            <Link
              id="beginGameButton"
              replace={false}
              to="/not-enough-characters"
            >
              <submitButton>
                Begin Game
              </submitButton>
            </Link>
          </section>
          <section
            id="grid"
          >
            <header>
              <h1>
                Uploaded Images
              </h1>
              <h5>
                Click Image to Remove
              </h5>
            </header>
          </section>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section>
              <h3>
                Upload 4 images that will be competing, (using a url)...
                <br />
                ...Or begin the game.
              </h3>
              <input
                default="image-url"
                onChange={[Function]}
                type="text"
              />
              <submitButton
                clicked={[Function]}
                id="uploadButton"
              >
                Upload
              </submitButton>
              <Link
                id="beginGameButton"
                replace={false}
                to="/not-enough-characters"
              >
                <submitButton>
                  Begin Game
                </submitButton>
              </Link>
            </section>,
            <section
              id="grid"
            >
              <header>
                <h1>
                  Uploaded Images
                </h1>
                <h5>
                  Click Image to Remove
                </h5>
              </header>
            </section>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Upload 4 images that will be competing, (using a url)...
                  <br />
                  ...Or begin the game.
                </h3>,
                <input
                  default="image-url"
                  onChange={[Function]}
                  type="text"
                />,
                <submitButton
                  clicked={[Function]}
                  id="uploadButton"
                >
                  Upload
                </submitButton>,
                <Link
                  id="beginGameButton"
                  replace={false}
                  to="/not-enough-characters"
                >
                  <submitButton>
                    Begin Game
                  </submitButton>
                </Link>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Upload 4 images that will be competing, (using a url)...",
                    <br />,
                    "...Or begin the game.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Upload 4 images that will be competing, (using a url)...",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "...Or begin the game.",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "default": "image-url",
                  "onChange": [Function],
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Upload",
                  "clicked": [Function],
                  "id": "uploadButton",
                },
                "ref": null,
                "rendered": "Upload",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <submitButton>
                    Begin Game
                  </submitButton>,
                  "id": "beginGameButton",
                  "replace": false,
                  "to": "/not-enough-characters",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Begin Game",
                  },
                  "ref": null,
                  "rendered": "Begin Game",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <header>
                  <h1>
                    Uploaded Images
                  </h1>
                  <h5>
                    Click Image to Remove
                  </h5>
                </header>,
                Array [],
              ],
              "className": undefined,
              "id": "grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1>
                      Uploaded Images
                    </h1>,
                    <h5>
                      Click Image to Remove
                    </h5>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Uploaded Images",
                    },
                    "ref": null,
                    "rendered": "Uploaded Images",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Click Image to Remove",
                    },
                    "ref": null,
                    "rendered": "Click Image to Remove",
                    "type": "h5",
                  },
                ],
                "type": "header",
              },
            ],
            "type": "section",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <div>
            <section>
              <h3>
                Upload 4 images that will be competing, (using a url)...
                <br />
                ...Or begin the game.
              </h3>
              <input
                default="image-url"
                onChange={[Function]}
                type="text"
              />
              <submitButton
                clicked={[Function]}
                id="uploadButton"
              >
                Upload
              </submitButton>
              <Link
                id="beginGameButton"
                replace={false}
                to="/not-enough-characters"
              >
                <submitButton>
                  Begin Game
                </submitButton>
              </Link>
            </section>
            <section
              id="grid"
            >
              <header>
                <h1>
                  Uploaded Images
                </h1>
                <h5>
                  Click Image to Remove
                </h5>
              </header>
            </section>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <section>
                <h3>
                  Upload 4 images that will be competing, (using a url)...
                  <br />
                  ...Or begin the game.
                </h3>
                <input
                  default="image-url"
                  onChange={[Function]}
                  type="text"
                />
                <submitButton
                  clicked={[Function]}
                  id="uploadButton"
                >
                  Upload
                </submitButton>
                <Link
                  id="beginGameButton"
                  replace={false}
                  to="/not-enough-characters"
                >
                  <submitButton>
                    Begin Game
                  </submitButton>
                </Link>
              </section>,
              <section
                id="grid"
              >
                <header>
                  <h1>
                    Uploaded Images
                  </h1>
                  <h5>
                    Click Image to Remove
                  </h5>
                </header>
              </section>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Upload 4 images that will be competing, (using a url)...
                    <br />
                    ...Or begin the game.
                  </h3>,
                  <input
                    default="image-url"
                    onChange={[Function]}
                    type="text"
                  />,
                  <submitButton
                    clicked={[Function]}
                    id="uploadButton"
                  >
                    Upload
                  </submitButton>,
                  <Link
                    id="beginGameButton"
                    replace={false}
                    to="/not-enough-characters"
                  >
                    <submitButton>
                      Begin Game
                    </submitButton>
                  </Link>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Upload 4 images that will be competing, (using a url)...",
                      <br />,
                      "...Or begin the game.",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Upload 4 images that will be competing, (using a url)...",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "...Or begin the game.",
                  ],
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "default": "image-url",
                    "onChange": [Function],
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Upload",
                    "clicked": [Function],
                    "id": "uploadButton",
                  },
                  "ref": null,
                  "rendered": "Upload",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <submitButton>
                      Begin Game
                    </submitButton>,
                    "id": "beginGameButton",
                    "replace": false,
                    "to": "/not-enough-characters",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Begin Game",
                    },
                    "ref": null,
                    "rendered": "Begin Game",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <header>
                    <h1>
                      Uploaded Images
                    </h1>
                    <h5>
                      Click Image to Remove
                    </h5>
                  </header>,
                  Array [],
                ],
                "className": undefined,
                "id": "grid",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h1>
                        Uploaded Images
                      </h1>,
                      <h5>
                        Click Image to Remove
                      </h5>,
                    ],
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Uploaded Images",
                      },
                      "ref": null,
                      "rendered": "Uploaded Images",
                      "type": "h1",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Click Image to Remove",
                      },
                      "ref": null,
                      "rendered": "Click Image to Remove",
                      "type": "h5",
                    },
                  ],
                  "type": "header",
                },
              ],
              "type": "section",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
